framework/src/wifi_manager/slsi_wifi_utils.c

```diff
@@ -120,6 +128,11 @@ fetch_scan_results(wifi_utils_scan_list_s **scan_list, slsi_scan_info_t **slsi_s
 			ndbg("num_sec modes(%d)\n", wifi_scan_iter->num_sec_modes);
 			ndbg("-----------------------------------------------\n");
 #endif
+			if (ssid != NULL && strlen(ssid) > 0 && strcmp(ssid, (const char *)wifi_scan_iter->ssid) != 0) {
```

Comment:
You'd better use `strncmp` instead of `strcmp`.

Reply:
Fixed, thanks